shell/platform/android/io/flutter/embedding/engine/android/AndroidTouchProcessor.java

```diff
@@ -86,8 +124,13 @@ public boolean onTouchEvent(MotionEvent event) {
   }
 
   // TODO(mattcarroll): consider creating a PointerPacket class instead of using a procedure that mutates inputs.
-  private void addPointerForIndex(MotionEvent event, int pointerIndex, ByteBuffer packet) {
-    int pointerChange = getPointerChangeForAction(event.getActionMasked());
+  private void addPointerForIndex(
```

Comment:
(optional) I know this is a private method so it's not strictly required, but I think documenting this would be helpful for people maintaining the class. I haven't looked deeply into this code before so could be just my unfamiliarity, but the params don't look obvious to me. Especially since `packet` is being mutated, but `event` isn't, and neither is marked as `final`.

Reply:
Unfortunately, I don't know more about this method than you.  Do you know what should be documented, or should we find a person who knows more?

Reply:
The thing that surprised me from the signature is that this is deliberately mutating the input param `packet` based on info from `event` and `pointerIndex`. It would be a little less surprising if that mutation was implied in the name or talked about in a comment describing the params.

I guess the `TODO` right above here does call this out though, so it's not really a big deal.

Reply:
Ok.  I agree it's surprising.  I was surprised, too, when I read it.  Let's improve in the near future.